src/utils.jl

```diff
@@ -45,16 +45,26 @@ Base.iterate(::Dimensions, ::Nothing) = nothing
 Base.iterate(q::Quantity) = (q, nothing)
 Base.iterate(::Quantity, ::Nothing) = nothing
 
-Base.zero(::Type{Quantity{T,R}}) where {T,R} = Quantity(zero(T), R)
+# Multiplicative identities:
 Base.one(::Type{Quantity{T,R}}) where {T,R} = Quantity(one(T), R)
-Base.one(::Type{Dimensions{R}}) where {R} = Dimensions{R}()
-
-Base.zero(::Type{Quantity{T}}) where {T} = zero(Quantity{T,DEFAULT_DIM_TYPE})
 Base.one(::Type{Quantity{T}}) where {T} = one(Quantity{T,DEFAULT_DIM_TYPE})
-
-Base.zero(::Type{Quantity}) = zero(Quantity{DEFAULT_VALUE_TYPE})
 Base.one(::Type{Quantity}) = one(Quantity{DEFAULT_VALUE_TYPE})
+Base.one(::Type{Dimensions{R}}) where {R} = Dimensions{R}()
 Base.one(::Type{Dimensions}) = one(Dimensions{DEFAULT_DIM_TYPE})
+Base.one(q::Quantity) = one(typeof(q))
```

Comment:
To be more explicit, should this throw an error pointing the user to `oneunit(::Quantity)` or `one(::Type{Quantity})` instead?

Reply:
(Though it is true that `q * one(q)` is perfectly valid...)

Reply:
This seems OK, although I don't love trivial `Quantity(1.0)`. It seems to me that if the units are known statically not to apply to something, it would be better not to have the type which allows for them. (It m/m cancels off, that's dynamic of course.)

What this routing via the type won't allow is `one(Quantity([1 2; 3 4], time=1))`.

Reply:
Oh, good point. Let me try to fix that.